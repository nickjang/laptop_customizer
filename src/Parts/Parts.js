import React, { Component } from 'react';
import slugify from 'slugify';
import USCurrencyFormat from '../USCurrencyFormat';

import './Parts.css'

class Parts extends Component {
  static defaultProps = {
    itemHash: '',
    feature: '',
    item: {
      name: '',
      cost: 0
    },
    selectedFeatureName: '',
    updateFeature: () => {return}
  };

  render() {
    return (
      <div className="feature__item">
        <input
          type="radio"
          id={this.props.itemHash}
          className="feature__option"
          name={slugify(this.props.feature)}
          checked={this.props.item.name === this.props.selectedFeatureName}
          onChange={() => this.props.updateFeature(this.props.feature, this.props.item)}
        />
        <label htmlFor={this.props.itemHash} className="feature__label">
          {this.props.item.name} ({USCurrencyFormat.format(this.props.item.cost)})
        </label>
      </div>
    );
  }
}

export default Parts;