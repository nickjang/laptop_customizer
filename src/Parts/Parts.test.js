import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Parts from './Parts';

describe('<Parts />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Parts />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders empty when no props supplied', () => {
    const wrapper = shallow(
    <Parts />
      );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders option when props are supplied', () => {
    const wrapper = shallow(
    <Parts key={1} itemHash="feature-1" feature="Display" item={{name:'15', cost:10}}
      selectedFeatureName={'10'}/>
      );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});