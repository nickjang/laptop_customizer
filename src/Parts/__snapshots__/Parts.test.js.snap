// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Parts /> renders empty when no props supplied 1`] = `
<div
  className="feature__item"
>
  <input
    checked={true}
    className="feature__option"
    id=""
    name=""
    onChange={[Function]}
    type="radio"
  />
  <label
    className="feature__label"
    htmlFor=""
  >
     (
    $0.00
    )
  </label>
</div>
`;

exports[`<Parts /> renders option when props are supplied 1`] = `
<div
  className="feature__item"
>
  <input
    checked={false}
    className="feature__option"
    id="feature-1"
    name="Display"
    onChange={[Function]}
    type="radio"
  />
  <label
    className="feature__label"
    htmlFor="feature-1"
  >
    15
     (
    $10.00
    )
  </label>
</div>
`;
